### ISSUE TITLE
Status report, future of the project

### ISSUE THREAD

**Issue by ArchangeGabriel** [Bumblebee-Project/Bumblebee#710]
Hi everyone here,

I’m aware we have been quite silent for a long time, because in fact I think that most of us moved away from the project.

I’m currently over busy but it will be better by the end of next week. At this point, I plan to read the 652 notifications from GitHub for this project that sits in my mailbox (plus the ones that will arrive in between), and try to answer to all of the currently remaining issues.

This will probably be quickly followed by a 4.0 release.

In the mean time, do not hesitate to ask for general things here.

Also, we will need to discuss the future of the project.

**Comment by gsgatlin** [comment]
So I noticed

https://blogs.gnome.org/uraeus/2015/08/19/fedora-workstation-next-steps-wayland-and-graphics/

(section where they talk about GL Dispatch)

and 

https://fedoraproject.org/wiki/Workstation/Tasklist

(Optimus support, Make Optimus setup work smoothly) 

I am skeptical if they can have something ready to fill in for bumblebee by fedora 24 time frame. (May 17, 2016) But who knows? They seem to get lots of complaints about optimus over at red hat. I'm pretty happy with bumblebee myself but I guess it would be cool if it could be more supported by the distro or GPU vendor. I don't know any of the individuals who are working on the optimus support stuff or any of the details.

Would be sad to see the bumblebee project end but I also understand if people have moved on to other interests.

**Comment by BenBlumer** [comment]
First of all: Thanks! Bumblebee has been a critical tool for me. I'm sure that's true for many others.

I'm not sure what the procedures are, but I would hope one of the bigger distros/companies would be willing to help you guys out. Your tool really is vital for people with Optimus laptops!

**Comment by Lekensteyn** [comment]
Thanks for starting this discussion @ArchangeGabriel.

At the moment my Nvidia GPU is always off with bbswitch, but my next laptop will get a GTX 965M (unfortunately). I do not need a discrete GPU, but since the card must be on for using an external monitor, I kind of need a working driver. I'll try hard to get nouveau working first. When I tried nouveau on a friends laptop with on an older GPU and external screen, it worked without issues.

With this new GPU, I might look again into some reported bbswitch issues. I think that some extra flags have to be passed to the DSM function, but I had not physically ran into this issue so ignored it for a while.

Personally I am not using bumblebeed, so there is no real dogfooding here. The core of Bumblebee is solid enough I think, recent reports are caused by nvidia blob issues (they have introduced some new modules which need to be unloaded via an appropriate modprobe conf).

**Comment by karolherbst** [comment]
sadly, nouveau can't handly that card, because there is signed firmware needed, which isn't released yet

**Comment by ArchangeGabriel** [comment]
@Lekensteyn @karolherbst is right, all 9xx NVIDIA GPU are not supported by nouveau because of the missing signed firmware.

With one week late, I’m starting to read every of the 737 notification emails that sit in my Bumblebee inbox and reply those that need to be answered, it should be done by the end of afternoon.

**Comment by ArchangeGabriel** [comment]
So, I’m basically done with bug triaging of bbswitch, a (quite) good summary of what is to be done is in https://github.com/Bumblebee-Project/bbswitch/issues/119.

Now going into Bumblebee.

**Comment by BenBlumer** [comment]
Missing signed firmware? Are you referring to the drivers or is this something else? What's the way to track this?

**Comment by ArchangeGabriel** [comment]
Yes, the nouveau driver is missing signed firmware from NVIDIA. You can either suscribe the nouveau-devel ML or just read Phoronix that will announce it on the day it will work, you can be sure of that. ;)

**Comment by smdrz** [comment]
can https://github.com/NVIDIA/libglvnd replace primus?

**Comment by ArchangeGabriel** [comment]
@smdrz Not straight away, but if Intel (mesa as a matter of fact) libgl happened to be translated to that, maybe you could just intercept calls at this level and change the vendor called. I’m not sure of all that, because I’m far from an expert in this area, but from what I understand it looks possible.

For Bumblebee 4.0, I’m scheduling this for beginning of may, I’m going to have some free time around there. Will have to go through 200 new messages, but should be able to do it. ;)

**Comment by danger89** [comment]
@ArchangeGabriel  Can I have write rights on the Bumblebee repo? Just to as back-up user. I will behave ;).

**Comment by ArchangeGabriel** [comment]
OK, I’m here now. @danger89: will discuss that with @Lekensteyn (just no to take such decision alone). Going to dig into the 284 bumblebee emails right now.

**Comment by danger89** [comment]
Ok no problem, I hope it won't take too long.

**Comment by ArchangeGabriel** [comment]
After discussion, we prefer to limit the number of people with access and are proposing rights to users with more involvement in/knowledge of the code. You’ve been doing a great job in the community and we hope you’ll continue to do so, but we don’t think a commit access would be useful. If that’s just for backup, we think that the addition of two new people, both very active in some upstream way, should be enough. ;) Hope you understand that. :)

@karolherbst and @bluca, do you accept to get those rights?

**Comment by bluca** [comment]
Always happy to help :-)

**Comment by danger89** [comment]
@ArchangeGabriel Ok, I understand. I am happy that I have caused this discussion. 
As a result, two new people will get added with these rights.
